primitive/src/main/java/io/atomix/primitive/partition/impl/PrimaryElectorService.java

```diff
@@ -310,10 +311,19 @@ ElectionState cleanup(Session session) {
                 .collect(Collectors.toList());
         if (primary.sessionId() == session.sessionId().id()) {
           if (!updatedRegistrations.isEmpty()) {
+            // find 1st backup to use as primary
+            Registration newPrimary = updatedRegistrations.get(0);
+            List<GroupMember> backups = term().backups(1);
```

Comment:
This change doesn’t make sense to me. The registrations are already sorted such that the primary is `registrations.get(0)` and the backups are `registrations.subList(1, numBackups)`. If we remove the primary (`registrations.remove(0)`) then the next primary (`registrations.get(0)`) is already a backup.

Reply:
The issue comes up when using a strategy like `MemberGroupStrategy.RACK_AWARE` for backup selection. In this case, the selected backups are not always `registrations.subList(1, numBackups)`, because the `PrimaryTerm.backups(int)` method may return different ones to avoid selecting two in the same group.

Reply:
I gotcha. This is indeed problematic. So the primary could be in e.g. group `1` and then fail over to a primary that's not actually used as a backup, thus potentially causing the backups to be recomputed as well if the new primary is in a different group. I think I need to think about this a bit...